xclim/indices/_hydrology.py

```diff
@@ -23,7 +27,7 @@
 
 
 @declare_units(q="[discharge]")
-def base_flow_index(q: xarray.DataArray, freq: str = "YS") -> xarray.DataArray:
+def base_flow_index(q: xr.DataArray, freq: str = "YS") -> xr.DataArray:
```

Comment:
What I will bring up was already there before the PR, but it would be good to address this while we're at it.

1. `discharge` is both a variable type defined in `units.py` (like "precipitation" & "temperature"), but also a variable name in `variables.yml` (like `pr` & `tas`). I think it should be one or the other
2. functions in `hydrology` should use something like `discharge` or `streamflow` instead of `q`, i.e. we should use a variable name defined in `variables.yml`
3. Speaking of which, `discharge` and `streamflow` in `variables.yml` seem to define the same quantity? Maybe we can remove `discharge` from `variables.yml` and keep it only as a variable type, then use `streamflow` as a variable name?

Reply:
I found this : https://github.com/Ouranosinc/xclim/pull/1431
So it seems the issue is that there is no convention on the name. `q` is often used, but I agree that its ambiguous.

I'm not sure (1) is a showstopper. But as  `discharge` is not actually used in any indicator, we could indeed remove it from the `variables.yml`. 

@RondeauG how breaking would it be for downstream (xhydro) if we changed all the `q` inputs to  `streamflow` ? Does that make sense to you ?

Reply:
>     3. Speaking of which, `discharge` and `streamflow` in `variables.yml` seem to define the same quantity? Maybe we can remove `discharge` from `variables.yml` and keep it only as a variable type, then use `streamflow` as a variable name?

This was because `discharge` was the legacy name used in `xclim`, but partners in `xhydro` much preferred `streamflow` (as `discharge` had a strong connotation with dams and regulated flow). 

However, I agree that `streamflow` is kinda long. `Q` is the standard in hydrology for streamflow in `m3 s-1`, and `q` for `m3 s-1 km-2`, but if it's too confusing with quantiles, we could find something else.

As to how breaking this would be for `xhydro`, this is completely fine for now. None of these indicators have been hard-coded anywhere in the package. The only explicit calls to `xclim` indicators are for  `stats` and the frequency analysis functions, both of which use `da`.

Reply:
If we sort out the situation with the variables used in the streamflow indices, then we can also address this comment ![here](https://github.com/Ouranosinc/xclim/pull/1881#discussion_r1727280353) and simplify the `Streamflow` class